### PR TITLE
Optimize citus connection management and cache some web3 repo methods

### LINKED ISSUE #8351: Web3 contract calls consume too many connections with citus db
### Description

Web3 `/contract/call` makes quite some heavy database queries, consumes a large number of db connections and causes degraded performance across all services dependent on a healthy database.

### Steps to reproduce

Check metrics, importer / rest / web3 logs in grafana cloud and triggered grafana alerts.

### Additional context

_No response_

### Hedera network

testnet

### Version

v0.104.0

### Operating system

None

### PR DESCRIPTION
**Pull request by xin-hedera**
**Description**:

This PR fixes the connection resource issue with citus and web3 in production environments

- Add caches to some web3 repository methods
- Change default pgbouncer pool_mode to `transaction` and keep `session` for `mirror_node` and `mirror_importer`
- Set a proper `max_prepared_statements` in pgbouncer
- Adjust the number of pgbouncer connections
- Adjust citus connection management parameters in favor of pgbouncer pooling
- Adjust citus max connections on both coordinators and workers
- Remove no longer needed session temp buffer size setting in importer

**Related issue(s)**:

Fixes #8351 

**Notes for reviewer**:

Perf comparison with 4 web3 pods and 80 concurrent users calling `/api/v1/contracts/call`
Note: it was tested with `ContractRepository.findRuntimeBytecode` cache, however I reverted the changes as it's duplicate work of Jesse's PR.

| Scenario    | RPS   | Avg. Response Time | Max number of web3 db connections to one worker |
|-------------|-------|--------------------|-------------------------------------------------|
| Unoptimized | 51/s  | 1.55               | 270                                             |
| Optimized   | 320/s | 30ms               | 85                                              |

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-mirror-importer/src/main/resources/application.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ hedera:
   mirror:
     importer:
       db:
-        connectionInitSql: set temp_buffers='256MB'; set timezone TO 'UTC';
+        connectionInitSql: set timezone TO 'UTC';
```
Comment (xin-hedera) [inline hedera-mirror-importer/src/main/resources/application.yml:5]: we no longer use temp tables in upsert

File: charts/hedera-mirror/values.yaml [hunk 2]
```diff
@@ -281,14 +281,14 @@ stackgres:
...
-      citus.max_shared_pool_size: "380"
+      citus.executor_slow_start_interval: "20ms"
+      citus.max_cached_conns_per_worker: "1"
+      citus.max_shared_pool_size: "850"
       full_page_writes: "false"  # Not needed with ZFS
       log_checkpoints: "true"
       log_timezone: "Etc/UTC"
       maintenance_work_mem: "256MB"
-      max_connections: "1200"
+      max_connections: "800"
```
Comment (xin-hedera) [inline charts/hedera-mirror/values.yaml:291]: we have two coord pods in active-active mode so it's actually 800 * 2 max connections

**Review by steven-sheehy — DISMISSED** [review]
LGTM

**Review by jnels124 — APPROVED** [review]
LGTM

**Review by jascks — DISMISSED** [review]
LGTM

**Review by jascks — APPROVED** [review]
LGTM

**Pull request merged**